hnn_core/network.py

```diff
@@ -832,6 +861,254 @@ def _get_src_type_and_pos(self, gid):
+
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        target_gids = self.gid_ranges[_long_name(target_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        target_types = np.repeat(target_cell, len(target_gids))
+
+        src_start = src_gids[0]  # Necessary for unique feeds
+        for target_type, target_gid in zip(target_types, target_gids):
+            if unique:
+                src_gids = [target_gid + src_start]
+                src_types = [src_types[0]]  # Assumes that all src_types match
+            for src_type, src_gid in zip(src_types, src_gids):
+                conn = {'src_type': src_type, 'src_gid': src_gid,
+                        'target_type': target_type, 'target_gid': target_gid,
+                        'loc': loc, 'receptor': receptor,
+                        'nc_dict': deepcopy(nc_dict)}
+                if not allow_autapses and src_gid == target_gid:
+                    continue
+
+                connectivity.append(conn)
+        connectivity_list.extend(connectivity)
+
+    def add_connection(self, src_gid, target_gid, loc, receptor, delay,
+                       weight, lamtha, threshold=0.0):
+        """Appends connections to connectivity list
+
+        Parameters
+        ----------
+        src_gid : int
+            Integer identifying source.
+        target_gid : int
+            Integer identifying target.
+        loc : str
+            Location of synapse on target cell. Must be
+            'proximal' or 'distal'.
+        receptor : str
+            Synaptic receptor of connection. Must be one of:
+            'ampa', 'nmda', 'gabaa', or 'gabab'.
+        delay : float
+            Time (ms) between source crossing threshold and
+            delivery of event to target.
+        weight : float
+            Synaptic weight on target cell.
+        lamtha : float
+            Space constant.
+        threshold : float
+            Source threshold.
+        """
+        conn = dict()
+        # Ensure gids in range of Network.gid_ranges
+        if not isinstance(src_gid, int):
+            raise TypeError(
+                "src_gid must be of type int, "
+                'got {}'.format(type(src_gid).__name__))
+
+        if not isinstance(target_gid, int):
+            raise TypeError(
+                "target_gid must be of type int, "
+                'got {}'.format(type(target_gid).__name__))
+
+        assert np.sum([src_gid in gid_range for
+                       gid_range in self.gid_ranges.values()]) == 1
+        assert np.sum([target_gid in gid_range for
+                       gid_range in self.gid_ranges.values()]) == 1
+
+        conn['src_gid'] = src_gid
+        conn['target_gid'] = target_gid
+        conn['src_type'] = self.gid_to_type(src_gid)
+        conn['target_type'] = self.gid_to_type(target_gid)
+
+        # Ensure string inputs
+        string_args = ['loc', 'receptor']
+        string_items = [loc, receptor]
```

Comment:
for 'loc' and 'receptor' don't you want to check that it is one of some valid options? like loc can only be 'distal' or 'proximal'

you can use this function: https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/externals/mne.py#L208